File: leetcode/easy/find-the-k-beauty-of-a-number.py
# https://leetcode.com/problems/find-the-k-beauty-of-a-number/

class Solution:
    def divisorSubstrings(self, num: int, k: int) -> int:
        s = str(num)
        return sum(int(s[i:i+k]) != 0 and num % int(s[i:i+k]) == 0 for i in range(len(s)-k+1))

File: leetcode/easy/find-the-k-beauty-of-a-number.ts
// https://leetcode.com/problems/find-the-k-beauty-of-a-number/

// MINE
function divisorSubstrings(num: number, k: number): number {
  const str = String(num)
  let ans = 0
  for (let i = 0; i <= str.length - k; ++i) {
    if (num % parseInt(str.substring(i, i + k)) === 0) ++ans
  }
  return ans
}

// ANSWER
